src/action_cc.cc

```diff
@@ -552,6 +554,177 @@ NM_ACTION_(nickel_wifi) {
     return nm_action_result_silent();
 }
 
+NM_ACTION_(nickel_orientation) {
+    Qt::ScreenOrientation o;
+    if      (!strcmp(arg, "portrait"))           o = Qt::PortraitOrientation;
+    else if (!strcmp(arg, "landscape"))          o = Qt::LandscapeOrientation;
+    else if (!strcmp(arg, "inverted_portrait"))  o = Qt::InvertedPortraitOrientation;
+    else if (!strcmp(arg, "inverted_landscape")) o = Qt::InvertedLandscapeOrientation;
+    else if (!strcmp(arg, "invert") || !strcmp(arg, "swap")) {
+        switch ((o = QGuiApplication::primaryScreen()->orientation())) {
+        case Qt::PrimaryOrientation:           NM_ERR_RET(nullptr, "could not get current screen orientation");                         break;
+        case Qt::PortraitOrientation:          o = !strcmp(arg, "invert") ? Qt::InvertedPortraitOrientation : Qt::LandscapeOrientation; break;
+        case Qt::LandscapeOrientation:         o = !strcmp(arg, "invert") ? Qt::InvertedLandscapeOrientation : Qt::PortraitOrientation; break;
+        case Qt::InvertedPortraitOrientation:  o = !strcmp(arg, "invert") ? Qt::PortraitOrientation : Qt::InvertedLandscapeOrientation; break;
+        case Qt::InvertedLandscapeOrientation: o = !strcmp(arg, "invert") ? Qt::LandscapeOrientation : Qt::InvertedPortraitOrientation; break;
+        default:                               NM_ERR_RET(nullptr, "unknown screen orientation %d", o);                                 break;
+        }
+    }
```

Comment:
I've implemented `invert` and `swap`.

Reply:
Side note: This could have been done in 2 lines with bit shifting and the `?:` ternary shorthand, but I chose readability :smiley:.

| Original | Invert | Swap |
| --- | --- | --- |
| 0001 | 0100 | 0010 |
| 0010 | 1000 | 0001 |
| 0100 | 0001 | 1000 |
| 1000 | 0010 | 0100 |